src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr)
+{
+    FILE *fPtr;
+    char buffer[256];
+    int foundSection = 0;
+    if (ParamPtr == NULL || SectionName == NULL || ParamName == NULL)
+        return M64ERR_INPUT_INVALID;
+    if (FileName == NULL || (fPtr = fopen(FileName, "rb")) == NULL)
+    {
+        DebugMessage(M64MSG_ERROR, "Unable to open config file '%s'.", FileName);
+        return M64ERR_INPUT_INVALID;
+    }
+    while (fgets(buffer, 255, fPtr) != NULL)
+    {
+        char *line = buffer;
+        ini_line l = ini_parse_line(&line);
+        switch (l.type)
+        {
+            case INI_SECTION:
+                if (osal_insensitive_strcmp(SectionName, l.name) == 0)
+                    foundSection = 1;
+                else
+                    foundSection = 0;
+                break;
+            case INI_PROPERTY:
+                if (foundSection)
+                {
+                    if (osal_insensitive_strcmp(ParamName, l.name) == 0)
+                    {
+                        fclose(fPtr);
+                        strncpy(ParamPtr, l.value, 256);
```

Comment:
This is dangerous in case of the dev didn't read the doc and pass a pointer to a tab whose size is less than 256 bytes.
One solution is to add an additional input parameter for the size of the input tab and pass it to strncpy.

But IMO the best solution is to do what @richard42 proposed and return a dynamically allocated string, the caller would be responsible for freeing the returned string.